qiita_db/metadata_template/test/test_prep_template.py

```diff
@@ -220,9 +220,11 @@ def test_get_none(self):
 class TestPrepSampleReadWrite(BaseTestPrepSample):
     """Tests the PrepSample class"""
     def test_setitem(self):
-        """setitem raises an error (currently not allowed)"""
-        with self.assertRaises(QiitaDBNotImplementedError):
-            self.tester['barcodesequence'] = 'GTCCGCAAGTTA'
+        with self.assertRaises(QiitaDBColumnError):
+            self.tester['column that does not exist'] = 0.3
+        self.assertEqual(self.tester['center_name'], 'ANL')
+        self.tester['center_name'] = "FOO"
+        self.assertEqual(self.tester['center_name'], "FOO")
```

Comment:
This also needs a test for when a column has the wrong datatype, triggering the ValueError.